peer.go

```diff
@@ -927,65 +971,120 @@ func (p *peer) handleLocalClose(req *closeLinkReq) {
+func (p *peer) handleResponseClosingSigned(msg *lnwire.ClosingSigned) {
 	p.activeChanMtx.RLock()
-	channel, ok := p.activeChannels[key]
+	channel := p.activeChannels[msg.ChannelID]
 	p.activeChanMtx.RUnlock()
-	if !ok {
-		peerLog.Errorf("unable to close channel, ChannelPoint(%v) is "+
-			"unknown", key)
-		return
-	}
 
-	// Now that we have their signature for the closure transaction, we
-	// can assemble the final closure transaction, complete with our
-	// signature.
-	sig := req.RequesterCloseSig
+	// verify signatures and mark channel closed
+	sig := msg.Signature
 	closeSig := append(sig.Serialize(), byte(txscript.SigHashAll))
 	closeTx, err := channel.CompleteCooperativeClose(closeSig)
 	if err != nil {
 		peerLog.Errorf("unable to complete cooperative "+
 			"close for ChannelPoint(%v): %v",
-			chanPoint, err)
+			channel.ChannelPoint(), err)
 		// TODO(roasbeef): send ErrorGeneric to other side
 		return
 	}
 
-	peerLog.Infof("Broadcasting cooperative close tx: %v",
-		newLogClosure(func() string {
-			return spew.Sdump(closeTx)
-		}))
+	closingTxid := closeTx.TxHash()
+	// Finally, launch a goroutine which will request to be notified by the
+	// ChainNotifier once the closure transaction obtains a single
+	// confirmation.
+	go p.waitForCloseConfirmation(channel, closingTxid, nil)
+}
 
-	// Finally, broadcast the closure transaction, to the network.
-	if err := p.server.lnwallet.PublishTransaction(closeTx); err != nil {
-		peerLog.Errorf("channel close tx from "+
-			"ChannelPoint(%v) rejected: %v",
-			chanPoint, err)
-		// TODO(roasbeef): send ErrorGeneric to other side
+// sendShutdown handles the creation and sending of the Shutdown messages
+// sent between peers to initiate the cooperative channel close workflow. In
+// addition, sendShutdown also signals to the HTLC switch to stop accepting
+// HTLCs for the specified channel.
+func (p *peer) sendShutdown(channel *lnwallet.LightningChannel) error {
+	_, addrs, _, err :=
+		txscript.ExtractPkScriptAddrs(channel.LocalDeliveryScript,
+			activeNetParams.Params)
+	if err != nil {
+		return err
+	}
+	address := lnwire.Address(addrs[0].ScriptAddress())
+
+	shutdown := lnwire.NewShutdown(*channel.ChannelPoint(), address)
+	p.queueMsg(shutdown, nil)
+
+	// Prevent the HTLC switch from receiving additional HTLCs for
+	// this channel.
+	p.server.htlcSwitch.UnregisterLink(p.addr.IdentityKey,
+		channel.ChannelPoint())
+
+	return nil
+}
+
+// sendClosingSigned handles the creation and sending of  proposed channel
+// close transactions.
+func (p *peer) sendClosingSigned(channel *lnwallet.LightningChannel) error {
+	// We agree with the proposed fee, so we send back our signature
+	// for the proposed transaction.
+	closeSig, err := channel.CreateCloseProposal()
+	if err != nil {
+		return err
+	}
+	channelPoint := channel.ChannelPoint()
+	closingSigned := lnwire.NewClosingSigned(*channelPoint, 5000, closeSig)
```

Comment:
The `5000` should be pulled out into a constant (and then later be made dynamic by your follow up work).